src/System.Net.Sockets/tests/FunctionalTests/UdpClientTest.cs

```diff
@@ -16,45 +16,344 @@ public class UdpClientTest
+            Assert.Throws<ObjectDisposedException>(() => udpClient.Connect(IPAddress.Loopback, 0));
+            Assert.Throws<ObjectDisposedException>(() => udpClient.Connect("localhost", 0));
+
+            Assert.Throws<ObjectDisposedException>(() => udpClient.Receive(ref remoteEP));
+
+            Assert.Throws<ObjectDisposedException>(() => udpClient.Send(null, 0, remoteEP));
+            Assert.Throws<ObjectDisposedException>(() => udpClient.Send(null, 0));
+            Assert.Throws<ObjectDisposedException>(() => udpClient.Send(null, 0, "localhost", 0));
+        }
+
+        [Fact]
+        public void Finalize_NoExceptionsThrown()
+        {
+            WeakReference<UdpClient> udpClientWeakRef = CreateAndDiscardUdpClient();
+
+            GC.Collect();
+            GC.WaitForPendingFinalizers();
+            GC.Collect();
+
+            UdpClient ignored;
+            Assert.False(udpClientWeakRef.TryGetTarget(out ignored));
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static WeakReference<UdpClient> CreateAndDiscardUdpClient() => new WeakReference<UdpClient>(new DerivedUdpClient());
+
+        [Fact]
+        public void Active_Roundtrips()
+        {
+            using (var udpClient = new DerivedUdpClient())
+            {
+                Assert.False(udpClient.Active);
+                udpClient.Active = true;
+                Assert.True(udpClient.Active);
+                udpClient.Active = false;
+                Assert.False(udpClient.Active);
+            }
+        }
+
+        [Fact]
+        public void Client_Idemptotent()
         {
-            UdpClient udpClient = new UdpClient(AddressFamily.InterNetwork);
-            byte[] sendBytes = new byte[1];
-            IPEndPoint remoteServer = new IPEndPoint(IPAddress.Loopback, UnusedPort);
+            using (var udpClient = new UdpClient())
+            {
+                Socket s = udpClient.Client;
+                Assert.NotNull(s);
+                Assert.Same(s, udpClient.Client);
 
-            Assert.Throws<ArgumentOutOfRangeException>("bytes", () =>
+                udpClient.Client = null;
+                Assert.Null(udpClient.Client);
+
+                udpClient.Client = s;
+                Assert.Same(s, udpClient.Client);
+
+                udpClient.Client = null;
+                s.Dispose();
+            }
+        }
+
+        [Fact]
+        public void Ttl_Roundtrips()
+        {
+            using (var udpClient = new UdpClient())
             {
-                udpClient.BeginSend(sendBytes, sendBytes.Length + 1, remoteServer, new AsyncCallback(AsyncCompleted), udpClient);
-            });
+                short ttl = udpClient.Ttl;
+                Assert.Equal(ttl, udpClient.Ttl);
+
+                udpClient.Ttl = 100;
+                Assert.Equal(100, udpClient.Ttl);
+            }
+        }
+
+        [PlatformSpecific(TestPlatforms.Windows)] // DontFragment not supported on Unix
```

Comment:
It throws a not supported exception, due to:
https://github.com/dotnet/corefx/blob/master/src/Native/Unix/System.Native/pal_networking.cpp#L2043
Maybe that's a bug rather than by design?  I'll open an issue and change this to ActiveIssue.